### PR TITLE
feat: swap improve

### PR DESCRIPTION
**Pull request by nuanyang233**
- feat: swap improve
- feat: swap trade info ui
- feat: price impact warning dialog

## Description

Closes #MF-470

## Type of change

- [ ] Documentation
- [x] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/extension/popups/pages/Wallet/ConnectWallet/index.tsx [hunk 1]
```diff
@@ -108,7 +113,7 @@ const ConnectWalletPage = memo(() => {
                 },
             })
         },
-        [isLocked, getLockStatusLoading],
+        [isLocked, getLockStatusLoading, wallets],
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/ConnectWallet/index.tsx]:
```suggestion
        [isLocked, getLockStatusLoading, wallets.length],
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx [hunk 2]
```diff
@@ -363,19 +365,19 @@ export function Trader(props: TraderProps) {
...
+                        open={openConfirmDialog}
+                        trade={focusedTrade.value}
+                        gas={focusedTrade.gas.value}
+                        gasPrice={gasPrice}
+                        inputToken={inputToken}
+                        outputToken={outputToken}
+                        onConfirm={onConfirm}
+                        onClose={onConfirmDialogClose}
+                    />
+                </>
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx]:
```suggestion
                <ConfirmDialog
                    open={openConfirmDialog}
                    trade={focusedTrade.value}
                    gas={focusedTrade.gas.value}
                    gasPrice={gasPrice}
                    inputToken={inputToken}
                    outputToken={outputToken}
                    onConfirm={onConfirm}
                    onClose={onConfirmDialogClose}
                />
```

File: packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingViewDeck.tsx [hunk 3]
```diff
@@ -42,6 +42,9 @@ const useStyles = makeStyles()((theme) => {
         content: {
             paddingTop: 0,
             paddingBottom: '0 !important',
+            ['&:last-child']: {
```
Comment (guanbinrui) [inline packages/mask/src/plugins/Trader/SNSAdaptor/trending/TrendingViewDeck.tsx]:
```suggestion
            '&:last-child': {
```

File: packages/mask/src/extension/popups/pages/Wallet/ConnectWallet/index.tsx (hunk 1 above)
Comment (UncleBill) [inline packages/mask/src/extension/popups/pages/Wallet/ConnectWallet/index.tsx]: Only care about whether there are wallets or not.

```suggestion
        [isLocked, getLockStatusLoading, !wallets.length],
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**